cura/Settings/AbstractMachine.py

```diff
@@ -14,14 +15,30 @@ def __init__(self, container_id: str) -> None:
         super().__init__(container_id)
         self.setMetaDataEntry("type", "abstract_machine")
 
-    def getMachines(self) -> List[ContainerStack]:
-        from cura.CuraApplication import CuraApplication
+    @classmethod
+    def getMachines(cls, abstract_machine: ContainerStack, online_only = False) -> List[ContainerStack]:
+        """ Fetches all container stacks that match definition_id with an abstract machine.
 
+        :param abstractMachine: The abstract machine stack.
+        :return: A list of Containers or an empty list if abstract_machine is not an "abstract_machine"
+        """
+        if not abstract_machine.getMetaDataEntry("type") == "abstract_machine":
+            return []
+
+        from cura.CuraApplication import CuraApplication  # In function to avoid circular import
         application = CuraApplication.getInstance()
         registry = application.getContainerRegistry()
 
-        printer_type = self.definition.getId()
-        return [machine for machine in registry.findContainerStacks(type="machine") if machine.definition.id == printer_type and ConnectionType.CloudConnection in machine.configuredConnectionTypes]
+        machines = registry.findContainerStacks(type="machine")
+        # Filter machines that match definition
+        machines = filter(lambda machine: machine.definition.id == abstract_machine.definition.getId(), machines)
+        # Filter only LAN and Cloud printers
+        machines = filter(lambda machine: ConnectionType.CloudConnection in machine.configuredConnectionTypes or ConnectionType.NetworkConnection in machine.configuredConnectionTypes, machines)
```

Comment:
combine lamda's

```suggestion
        def machinesFilter(machine):
            if machine.definition.id is not abstract_machine.definition.getId():
                return false
            return ConnectionType.CloudConnection in machine.configuredConnectionTypes or ConnectionType.NetworkConnection in machine.configuredConnectionTypes
        machines = filter(machinesFilter, machines)
```
note: not tested

Reply:
I intentionally have them separated. I think it makes it easier to understand the filtering steps.